pkg/apis/seedmanagement/helper/helper.go

```diff
@@ -32,28 +34,37 @@ func GetBootstrap(bootstrap *seedmanagement.Bootstrap) seedmanagement.Bootstrap
 
 // IsMultiZonalManagedSeed checks if a managed seed is multi-zonal.
 func IsMultiZonalManagedSeed(managedSeed *seedmanagement.ManagedSeed) (bool, error) {
-	seedSpec, err := ExtractSeedSpec(managedSeed)
+	seedTemplate, _, err := ExtractSeedTemplate(managedSeed)
 	if err != nil {
 		return false, err
 	}
-	return seedSpec.HighAvailability != nil && seedSpec.HighAvailability.FailureTolerance.Type == gardencore.FailureToleranceTypeZone, nil
+	return seedTemplate.Spec.HighAvailability != nil && seedTemplate.Spec.HighAvailability.FailureTolerance.Type == gardencore.FailureToleranceTypeZone, nil
 }
 
-// ExtractSeedSpec extracts the seed spec from the ManagedSeed.
-func ExtractSeedSpec(managedSeed *seedmanagement.ManagedSeed) (*gardencore.SeedSpec, error) {
+// ExtractSeedTemplate extracts the seed template from ManagedSeed along with the path for the SeedTemplate.
+func ExtractSeedTemplate(managedSeed *seedmanagement.ManagedSeed) (*gardencore.SeedTemplate, *field.Path, error) {
 	if managedSeed.Spec.SeedTemplate != nil {
-		return &managedSeed.Spec.SeedTemplate.Spec, nil
+		return managedSeed.Spec.SeedTemplate, field.NewPath("spec", "seedTemplate"), nil
 	}
-	// Check if gardenlet has been set and get the SeedTemplate from it
+	gardenletConfig, err := GetGardenletConfiguration(managedSeed)
+	if err != nil {
+		return nil, nil, err
+	}
+	if gardenletConfig != nil && gardenletConfig.SeedConfig != nil {
+		return &gardenletConfig.SeedConfig.SeedTemplate, field.NewPath("spec", "gardenlet", "config", "seedConfig"), nil
+	}
+	return nil, nil, fmt.Errorf("no seed template found for managedseed %s", managedSeed.Name)
+}
+
+// GetGardenletConfiguration converts and gets the config.GardenletConfiguration from the ManagedSeed if one is defined.
+func GetGardenletConfiguration(managedSeed *seedmanagement.ManagedSeed) (*config.GardenletConfiguration, error) {
```

Comment:
Make this function rather private? Or please add unit tests.

Reply:
I will make it private for now as it is not been used elsewhere. This method is already tested indirectly using tests written for `ExtractSeedTemplate`

Reply:
pushed the changes.